### PR TITLE
Specifying torch dtype in Qwen2VLForConditionalGeneration

### PR DESCRIPTION
**Pull request by htahboub**
Fixing a bug that led to the warning

```You are attempting to use Flash Attention 2.0 without specifying a torch dtype. This might lead to unexpected behaviour```

every time this model was initialized with `attn_implementation="flash_attention_2"`, even with the dtype specified.

@amyeroberts, @qubvel

### CODE REVIEW

**Comment by qubvel** [comment]
cc @zucchini-nlp if you have bandwidth

**Review by zucchini-nlp — COMMENTED** [review]
Thanks for the PR!

I don't think the proposed fix is a good way as it would prompt us to pass a "dtype" when init the backbone in all composite models (a lot of code change) and I haven't seen us using `config.dtype`. Usually the dtype is set when loading the model with `torch.get_default_dtype()`. 

I have a huuge PR to make attn implementation API more consistent with what is expected for composite models in https://github.com/huggingface/transformers/pull/32238. That should fix the issue with warning, but I am also okay to merge separately the PR related to `dtypes`. My personal preference for fix would be to tweak `modeling_utils.py` file to get the default dtype (as fallback) when loading from config

**Comment by htahboub** [comment]
@zucchini-nlp Ah I see, agreed in that case. And I agree that changing the fallback from `None` to the default dtype would be a lot cleaner, so I'll make that change to `modeling_utils.py` then and revert the current proposed fix.

**Comment by htahboub** [comment]
Hey @zucchini-nlp, since this is a pretty easy fix for the bug, I suppose we should either merge now or close this PR since the change is already included in your big PR. So, if you think your PR is about to be merged soon, please let me know so we can just close this, or if it's gonna take a while longer, then I think we'd be better off merging this as soon as possible because the warning with Qwen can be a bit confusing to people encountering it. Thanks!

**Review by zucchini-nlp — APPROVED** [review]
Oh I didn't see you made the changes already. Yes, the linked PR will take a while as it touches so many models and opened a can of worms. 

LGTM, agree that we should merge this. After the cpre maintainer approves, we'll merge it :)

**Review by ArthurZucker — APPROVED** [review]

**Pull request merged**
